src/coreclr/jit/flowgraph.cpp

```diff
@@ -5998,38 +6062,193 @@ void FlowGraphNaturalLoop::Duplicate(BasicBlock** insertAfter, BlockToBlockMap*
 {
     assert(CanDuplicate(nullptr));
 
-    Compiler* comp = m_dfsTree->GetCompiler();
+    Compiler* const   comp           = m_dfsTree->GetCompiler();
+    bool              canCloneTry    = false;
+    bool              clonedTry      = false;
+    BasicBlock* const insertionPoint = *insertAfter;
+
+    INDEBUG(canCloneTry = (JitConfig.JitCloneLoopsWithEH() > 0);)
```

Comment:
I think this should be a parameter. Loop cloning is not the only user of this utility.

Reply:
Indeed. Unrolling gave me more problems than cloning, and unrolling loops with EH is likely almost never a good idea, or perhaps should be handled differently (say only unroll if we can make the try region cover all the unrolls).